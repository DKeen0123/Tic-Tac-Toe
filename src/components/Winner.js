import React from 'react';

const Winner = () => {
  return (
    <div>
      <h1>Player 1 wins!</h1>
    </div>
  );
};

export default Winner;
