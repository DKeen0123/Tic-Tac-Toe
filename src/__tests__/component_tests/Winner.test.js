import React from 'react';
import { shallow } from 'enzyme';
import Winner from '../../components/Winner';

describe('Winner', () => {
  let props = { winningPlayer: 'Player 1' };
  let winner = shallow(<Winner {...props} />);

  it('renders correctly', () => {
    expect(winner).toMatchSnapshot();
  });

  describe('when player 1 wins', () => {
    it('displays that player 1 wins', () => {
      expect(winner.text()).toEqual('Player 1 wins!');
    });
  });

  describe('when player 2 wins', () => {
    it('displays that player 2 wins', () => {
      props = { winningPlayer: 'Player 2' };
      winner = shallow(<Winner {...props} />);
      expect(winner.text()).toEqual('Player 2 wins!');
    });
  });
});
