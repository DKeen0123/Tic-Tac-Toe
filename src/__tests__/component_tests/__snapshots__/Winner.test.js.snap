// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Winner renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Winner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Player 1 wins!
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Player 1 wins!",
      },
      "ref": null,
      "rendered": "Player 1 wins!",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Player 1 wins!
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Player 1 wins!",
        },
        "ref": null,
        "rendered": "Player 1 wins!",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
